text/0000-enable-embroider.md

```diff
@@ -0,0 +1,190 @@
+
+### Packages
+
+The following packages will be added to package.json:
+
+- @embroider/compat - Stage 1 is responsible for upgrading addons from their current implmentations ("V1") into a new format called "V2 Packages". This package also contains the entry point method that is invoked in your ember-cli-build.js file (shown below).
+- @embroider/core - Stage 2 is responsible for taking a collection of V2 packages and abstracting away the "emberness" into standard Javascript which all packagers can understand.
+- @embroider/webpack - Stage 3 of Embroider's design allows you to bring your own packager. By default Embroider ships with and recommends using Webpack
+
+### Build File Changes:
+
+Embroider replaces the logic provided by `ember-app.js`. By default it runs in a "safe" mode that provides the best backwards compatibility guarantees but prevents some optimizations from happening. This safe mode configuration is what we will ship by default:
+
+```diff
+-return app.toTree();
++const { Webpack } = require('@embroider/webpack');
++return require('@embroider/compat').compatBuild(app, Webpack);
+```
+
+### New application flow
+
+The above changes will be applied to the blueprint for `ember new`. Additionally, the build flag option of `--embroider` will exist so that the option will be recorded in `config/ember-cli-update.json` like:
+
+```json
+{
+  "schemaVersion": "1.0.0",
+  "packages": [{
+    "name": "ember-cli",
+    "version": "X.X.X",
+    "blueprints": [{
+      "name": "app",
+      "outputRepo": "https://github.com/ember-cli/ember-new-output",
+      "codemodsSource": "ember-app-codemods-manifest@1",
+      "isBaseBlueprint": true,
+      "options": [
+        "--embroider"
+      ]
+    }]
+  }]
+}
+```
+
+This will ensure that `ember-cli-update` only applies these changes to applications that are already on Embroider.
+
+### Existing application migration flow
+
+In order to make the migration as simple as possible, a codemod will be created. The codemod CLI will provide two commands: `preflight` and `migrate`.
+
+Example usage:
+
+`npx @embroider/cli preflight`
+`npx @embroider/cli migrate`
+
+#### Preflight
+
+The goal of this command is to give users an easy yes or no answer to the question: can I start migrating to Embroider? As an example, we can check if the project is on the minimal set of package versions of known "fixed" addons along with other static checks.
+
+#### Migrate
+
+The migrate command is purely to codemod the file changes above to get your app onto Embroider.
+
+### Deprecations
+
+While Embroider strives to be as backwards compatible as possible, some existing concepts will not directly apply under Embroider. These concepts largely rely around the fact that Embroider does not want to you care about the contents of your output. This means that concepts like fingerprints asset file names or modifing output paths of your dist assets will not be supported.
+
+#### Fingerprinting
```

Comment:
Bringing over a comment thread from a [predraft version](https://github.com/thoov/rfcs/pull/1) of this RFC: 

@kategengler: 
Are other fingerprint options still supported?
 
@thoov:
Currently no. I can make this section state more explicitly that app.options.fingerprint are not enabled
 
@simonihmig:
I think we could (should?) point out that there is still quite a gap in the functionality, as Embroider will only fingerprint the basic JS and CSS files that it adds to index.html. broccoli-asset-rev would fingerprint arbitrary files, e.g. images, and rewrite all the references to those (as long as it can "see" the file paths, so they have to be literals, i.e. no dynamic concatenations logic or so), e.g in JS or CSS files. AFAIK there is no solution for this yet in Embroider-land.

Which is quite annoying as you cannot serve images with HTTP headers that enable long-term browser caching as you normally would. Or if you do, they are not invalidated when they change.

And people are probably quite used to this (the previous behavior), so when this does not work it would feel like a bug rather than a missing feature.